src/js/services/image.service.test.js

```diff
@@ -104,17 +104,18 @@ describe('getImagesService', function(){
                     var farm, server, id, title, secret, picUrl, currentPhoto;
                     var images = [];
 
-                    for (var i = 0; i < data.photos.photo.length; i++) {
-                        currentPhoto = data.photos.photo[i];
-                        farm = currentPhoto.farm;
-                        server = currentPhoto.server;
-                        id = currentPhoto.id;
-                        title = currentPhoto.title;
-                        secret = currentPhoto.secret;
+                   angular.forEach(data.photos.photo, function (photo, index) {
+
+                        farm = photo.farm;
+                        server = photo.server;
+                        id = photo.id;
+                        title = photo.title;
+                        secret = photo.secret;
                         picUrl = "https://farm" + farm + ".staticflickr.com/" + server + "/" + id + "_" + secret + "_z.jpg";
+
                         images.push(picUrl);
-                        expect(images[i]).to.equal("https://farm" + farm + ".staticflickr.com/" + server + "/" + id + "_" + secret + "_z.jpg");
-                    }
+                        expect(images[index]).to.equal("https://farm" + farm + ".staticflickr.com/" + server + "/" + id + "_" + secret + "_z.jpg");
+                    });
                 };
 
                 createImagesArray(sampleJSON);
```

Comment:
I had to look at the entire file for this one....

I think this test should be revisited. You declare a function in the test, then call it and check that it's working. Ideally, we want the actual code to be doing the heavy lifting, our tests are just verifying that the result of that function is correct.